app/src/main/java/org/p2p/wallet/receive/tokenselect/ReceiveTokensPresenter.kt

```diff
@@ -40,22 +53,31 @@ class ReceiveTokensPresenter(
                 secondTokenUrl = ethToken?.iconUrl.orEmpty()
             )
             observeTokens()
+            observeMappedTokens()
         }
     }
 
     override fun load(isRefresh: Boolean, scrollToUp: Boolean) {
         launch {
+            if (isRefresh) {
+                tokensFlow.value = emptyList()
+            }
             this@ReceiveTokensPresenter.scrollToUp = scrollToUp
             interactor.fetchTokens(searchText, PAGE_SIZE, isRefresh)
         }
     }
 
     override fun onSearchTokenQueryChanged(newQuery: String) {
-        searchText = newQuery
-        if (newQuery.isBlank()) {
-            view?.resetScrollPosition()
+        searchJob?.cancel()
+        searchJob = launch {
+            delay(300L)
```

Comment:
Noooo :( not delay :(

Reply:
delay here is totally good, just move it to const please